### DISCUSSION TITLE
0.8.0

### DISCUSSION THREAD

**Discussion by darkskygit** [toeverything/OctoBase#531] [Announcements]
### OctoBase

- S3 Support for Blob Storage @thorseraq @Xuanwo (#466, #473, #474)
- Workspace root doc GUID should be consistent @forehalo (#480)
- Make workspace root doc's guid the same as workspaceId @thorseraq (#483)
- Refactor Core and RPC @darkskygit  (#494, #503)
- Make storage API friendly @darkskygit @forehalo (#472, #489, #527, #528)

### [Y-Octo](https://github.com/toeverything/y-octo)

- Thread safety improvements @darkskygit @forehalo @zxch3n (toeverything/y-octo#9, #496, #512, #518, #520)
- Memory management improvements @darkskygit @forehalo (toeverything/y-octo#16, #478, #488, #522, #525)
- Binary codec improvements @forehalo @thorseraq (#475, #484, #497, #504)
- CRDT operation correctness improvements @darkskygit @forehalo (#495, #506, #516, #519, #524, #526)

### Tests

1. Improve code robustness @darkskygit (toeverything/y-octo#10, #470, #485, #502, #517)
2. Benchmarking improvements @darkskygit (toeverything/y-octo#11, #468)

## TLDR

In this release, we've validated our CRDT implementation, [jwst-codec](https://github.com/toeverything/OctoBase/tree/0be9fb3074daedfc75be22c574e1484d9fb5edb7/libs/jwst-codec), across a multitude of production scenarios. We're pleased to announce that it has been operating stably in the AFFiNE Cloud production environment for over a month. We've encapsulated this CRDT functionality into a standalone crate: [Y-Octo](https://github.com/toeverything/y-octo). Y-Octo is a tiny, ultra-fast CRDT collaboration library and has interoperability and binary compatibility with [yjs](https://github.com/yjs/yjs).

Additionally, we have continued to make enhancements to OctoBase. Key updates in this release include the integration of S3 as a supported storage target (#466, #473, #474), along with improvements to our storage API (#472, #489, #527, #528).

## New Contributors
* @Xuanwo made their first contribution in https://github.com/toeverything/OctoBase/pull/473
* @zxch3n made their first contribution in https://github.com/toeverything/y-octo/pull/9

**Full Changelog**: https://github.com/toeverything/OctoBase/compare/p20230614...p20230904

<hr /><em>This discussion was created from the release <a href='https://github.com/toeverything/OctoBase/releases/tag/p20230904'>0.8.0</a>.</em>
